lessons/misc/project-management-R/lesson.md

```diff
@@ -0,0 +1,230 @@
+
+To start a new `Rproject`, navigate to `file > new project` and choose `existing directory`. Choose the `project-management-demo` directory that is part of this tutorial. Create the `Rproject`.
+
+In the `files` pane in `Rstudio`, you should now see a `project-management-demo.Rproj` file. This `.Rproj` file contains the preferences for your `Rproject`. This is also the file that you and collaborators would click on to open the existing project, with all preferences maintained across platforms and computers.
+
+One of the best things about `Rprojects` is that the working directory is automatically set to the directory containing the `.Rproj` file, regarless of which computer the file is opened on. We can confirm this now by typing the following the the **R console**.
+
+```
+getwd()
+```
+
+Notice the working directory is now the current project's location on your computer. This allows us to access files and run scripts without constantly having to switch directories.
+
+### Initializing `Git`
+
+Before moving on, lets initialize this project as a git repo. From your shell, make sure you are in the root of the project, then type:
+
+```
+git init
+```
+
+You have now started a git repository. Lets check the status of our repo.
+
+```
+git status
+```
+
+Git is telling us that it isn't tracking anything (as expected). In fact, there are a few files that we'll probably never want git to track. Let's add those to a file called `.gitignore` now.
+
+```
+vi .gitignore
+```
+
+Add `.Rhistory` and `.Rproj.user/` on separate lines in the file (Make sure you're in insert more by first typin `i`). Then, quit using `:wq`. Note you can add any other directories or files you don't want git to track directly to this file.
+
+Check the status again by typing `git status`. You should no longer see the `.Rhistory` and `.Rproj.user/` file, but instead see the `.gitignore` file. Lets commit these changes.
+
+```
+git add .
+git commit -m "Initial commit"
+```
+
+Another `git status` confirms that the repo is up to date and git hasn't recognized any changes.
+
+### Initializing `Packrat`
+
+Now that `git` is initialized and tracking the files in this directory, we can initialize `packrat`, which will keep track of installed packages and their versions. This ensures that if you come back to the project in two years, after having updated many packages, your analyses will still be based on the versions you were initially using.
+
+Lets click on the `.Rproj` file to open the `Rproject` options. In the left sidebar, click on `Packrat` and then on `use packrat with this project`. We'll want to check both `git ignore packrat library` and `git ignore package sources`.
```

Comment:
My RStudio seems to be missing these last two options

Reply:
Also, packrat seems to take several minutes to check packages if you have many installed - could be worth mentioning that?

Reply:
Failed the first time - had to rerun `packrat::snapshot()` and then it worked